File: csharp/MaxPainInfrastructure/Models/Hop.cs
namespace MaxPainInfrastructure.Models
{
    public class Hop
    {
        public long Id { get; set; }
        public string? Destination { get; set; }
        public string? Referrer { get; set; }
        public string? UserAgent { get; set; }
        public DateTime? CreatedOn { get; set; }
    }

    public class HopSummary
    {
        public DateTime? CreatedOn { get; set; }
        public string? Destination { get; set; }
        public int Hops { get; set; }
    }

    public class HopAgent
    {
        public string? UserAgent { get; set; }
        public int Hops { get; set; }
    }

}

File: csharp/MaxPainUI/ClientApp/src/app/models/hop.ts
import { Pipe, PipeTransform } from '@angular/core';
import { DatePipe } from '@angular/common';
import { UtilsService } from '../services/utils.service';

export class Hop {
  id: Number;
  destination: String;
  referrer: String;
  createdOn: Date;

  constructor(private utils: UtilsService) { }

  get est(): Date {
    return this.utils.utcToEst(this.createdOn);
  }
}

export class HopSummary {
    createdOn: Date;
    destination: String;
    hops: number;
  
    constructor(private utils: UtilsService) { }
}
  
export class HopAgent {
  userAgent: String;
  hops: number;

  constructor(private utils: UtilsService) { }
}
